game/script-topics.rpy

```diff
@@ -599,6 +599,573 @@ label talk_eating_well:
+            n 1nnmaj "I hope you didn't forget to confirm your email address too!"
+            n 1nnmss "Alright!{w=0.2} Now here's the challenging part."
+
+        "I already had an account set up.":
+            n 1uwdaj "Oh?{w=0.2} I didn't realise you were such a pro at this,{w=0.1} [player]!{w=0.2} Ehehe."
+
+            if already_discussed_setup:
+                n 1nlrdv "Or maybe we went through this before,{w=0.1} hmm?" #Not quite sure with this expression
+
+            n 1nnmbg "Well, anyway.{w=0.2} The rest of this should be a piece of cake then!"
+
+        "Never mind.":
+            n 1tnmpu "Huh?{w=0.2} You don't wanna continue?"
+            n 1nlraj "That's fine,{w=0.1} I guess."
+            n 1nnmss "Just let me know when you're ready,{w=0.1} 'kay?"
+            return {"lock" : None}
+
+    # Prompt the player for an API key
+
+    n 1nnmbg "Are you ready,{w=0.1} [player]?"
+    n 1nllss "You need to get your API key and send it to me!"
+    n 1nnmbg "You can find your keys {a=[store.jn_globals.LINK_OPEN_WEATHER_MAP_API_KEYS]}here{/a},{w=0.1} or you can get there using the menu like before."
+    n 1nwrts "You should have one set up already,{w=0.1} but you can create another just for me if you want!{w=0.2} Ehehe."
+    n 1nchbg "Don't worry, I won't make you type it all up. You can just paste it,{w=0.2} hehe. "
+    n 1nnmss "{a=[store.jn_globals.LINK_OPEN_WEATHER_MAP_API_KEYS]}Here's{/a} the link in case you forgot{w=0.1} -{w=0.1} and again,{w=0.1} just talk to me when you have it ready!"
+    n 1nnmsm "..."
+    n 1nwdbg "Okaaay!{w=0.2} Take it away,{w=0.1} [player]!"
+
+    $ player_input_valid = False
+    $ player_input_count = 0
+
+    # Process the player's input
+
+    while not player_input_valid:
+
+        if player_input_count >= 3:
+            # Escape route if we somehow get stuck
+            n 1kllaj "I...{w=0.3} don't think we're getting anywhere with this,{w=0.1} [player]."
+            n 1nnmss "Maybe we should just try again later?"
+            return {"lock" : None}
+
+        else:
+            $ player_input = renpy.input("Enter your API key (or type Nevermind to go back):")
+
+        if not player_input or player_input == "":
+            n 1nlraj "Uh...{w=0.3} Did you actually type anything,{w=0.1} [player]?"
+            n 1nwmss "I didn't get that...{w=0.3} can you try again for me?"
+            $ player_input_count += 1
+
+        elif player_input.replace(" ", "").lower() == "nevermind":
+            # Allow the player to back out
+            n 1knmaj"Huh?{w=0.2} You don't wanna continue?"
+            n 1nwmpu "That's fine,{w=0.1} I guess."
+            n 1nnmss "Just let me know when you're ready,{w=0.1} 'kay?"
+            return {"lock" : None}
+
+        else:
+            # Get ready to lead in to the next stage of setup
+            $ player_input_valid = True # Kill the loop!
+            $ persistent.weather_api_key = player_input
+            n 1nnmlg "Alright!{w=0.2} I got it!"
```

Comment:
Once more, I feel like closed eyes and raised eyebrows would work a little better here.